gobblin-data-management/src/main/java/gobblin/data/management/copy/CopyableFile.java

```diff
@@ -103,10 +109,12 @@
    * @param origin {@link FileStatus} of the original file.
    * @param datasetRoot Value of {@link CopyableDataset#datasetRoot} of the dataset creating this {@link CopyableFile}.
    * @param copyConfiguration {@link CopyConfiguration} for the copy job.
+   * @param originTimestamp Timestamp of file at its origin source.
+   * @param upstreamTimestamp Timestamp of file as in upstream.
    * @return a {@link CopyableFile.Builder}.
    */
   public static Builder builder(FileSystem originFs, FileStatus origin, Path datasetRoot,
-      CopyConfiguration copyConfiguration) {
+      CopyConfiguration copyConfiguration, long originTimestamp, long upstreamTimestamp) {
```

Comment:
it should not be necessary to have the timestamps as arguments, that is what the builder is for.